src/CMakeVersion.txt

```diff
@@ -0,0 +1,14 @@
+set(GIT_CMD "git")
+set(GIT_ARGS "log" "-n" "1" "--pretty=format:%h - %ad")
+execute_process(COMMAND ${GIT_CMD} ${GIT_ARGS}
+    RESULT_VARIABLE GIT_RESULT
+    OUTPUT_VARIABLE GIT_COMMIT)
+    if (GIT_RESULT)
+        set (GIT_COMMIT "UNKNOWN")
+    endif()
+set(ZMAP_VERSION "Development Build. Commit ${GIT_COMMIT}")
+
+configure_file(topt.ggo.in topt.ggo @ONLY)
```

Comment:
1) I did my best to patch this up, but since out of source builds are broken elsewhere, it's not fully tested. But I think it will work if we fix the top-level issues.

2) Ah. Yes, that's not impacted since we're doing per-build generation when in development mode.

3) Let's decouple the cmake 'add_custom_WHAT` from what it invokes (since both command and target can invoke either).

Between `add_custom_target` and `add_custom_command`, target seems better since you want this invoked every build. That's what `add_custom_target` is built for. What dependency would you specify for `add_custom_command`? Seems like we'd need to log the commit in some file somewhere and add per-build hooks to check the value. Since both require per-build hooks, skipping writing state to a file somewhere seems preferable.

RE what to invoke: Since we're replicating what the build system is already doing (replacing a cmake variable in a file) reusing the build system seemed preferable. I also like not having bash scripts laying around in the project. Do you care enough to request this be scrapped and replaced with a bash script?  (The current solution seems complete, not withstanding style comments.)